### DISCUSSION TITLE
brew services start ollama is giving an error (Permission denied @ rb_sysopen)

### DISCUSSION THREAD

**Discussion by gtnbssn** [Homebrew/discussions#4787] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.11-17-g24de357
ORIGIN: https://github.com/Homebrew/brew
HEAD: 24de357fbf5901d950cfa73acbffcbb93b4aba7f
Last commit: 2 days ago
Core tap JSON: 13 Sep 10:55 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.5.2-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

[ollama](https://github.com/jmorganca/ollama) is an app that makes it extremely easy to run LLMs locally.

The formula to install it is there: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/o/ollama.rb

After running `brew install ollama`, the user gets a message that says:

```
==> Caveats
To start ollama now and restart at login:
  brew services start ollama
Or, if you don't want/need a background service you can just run:
  /opt/homebrew/opt/ollama/bin/ollama serve
```

But running `brew services start ollama` gives the following error:

`Error: Permission denied @ rb_sysopen - /Users/gtnbssn/Library/LaunchAgents/homebrew.mxcl.ollama.plist`

The ollama maintainers recommend downloading and installing directly from [their website](https://ollama.ai/) instead (the app will take care of updating itself). But I was wondering if something could be fixed in homebrew, or maybe something is wrong in my system. I like the idea that I am managing all my apps through the same package manager.

**Comment by SMillerDev** [comment]
It seems you don't have permission to write to that file. You should check what `ls -al FILENAME` says

**Reply by gtnbssn** [reply]
Well it turns out the file doesn't even exist!

**Reply by gtnbssn** [reply]
Could be that the formula is missing some of the things explained here: https://docs.brew.sh/Formula-Cookbook#service-files

**Reply by SMillerDev** [reply]
How about `ls -al /Users/gtnbssn/Library/LaunchAgents/` though?

**Reply by gtnbssn** [reply]
```
total 8
drwxr-xr-x   3 root     staff    96 Apr 28 11:07 ./
drwx------@ 90 gtnbssn  staff  2880 Aug 11 17:25 ../
-rw-r--r--   1 gtnbssn  staff   616 Apr 28 11:07 net.pulsesecure.SetupClient.plist
```

So nothing homebrew related in there. I have two things homebrew has registered as services though:

```
> brew services list                                                                                                                                         ✔  ⬢ 18.16.0 
Name    Status User File
ollama  none        
unbound none        
```

**Reply by SMillerDev** [reply]
The folder is owned by root, that's most likely why it can't write to it.

**Reply by gtnbssn** [reply]
Is this folder not supposed to be owned by root under MacOS?

According to [this](https://discussions.apple.com/thread/7018347) it shouldn't. Which makes a lot of sense. Maybe it was created by a process I ran with sudo?

I see I do have a service in there, for something I guess required me to enter my password when I installed it. Should brew install for this formula ask for my password to setup the service?

Sorry, really not sure how things are supposed to work here!

**Reply by SMillerDev** [reply, marked as answer]
> Is this folder not supposed to be owned by root under MacOS?

Nothing your user's folder is supposed to be owned by root on any OS.

> Should brew install for this formula ask for my password to setup the service?

It should note if it's installing in a location that should be fully owned by you

**Reply by gtnbssn** [reply]
Thanks! I chown'ed it back to my username. Not sure how this happened, maybe I abused sudo at some point...
